src/wallet/cbf.rs

```diff
@@ -0,0 +1,88 @@
+use std::{collections::HashMap, net::SocketAddr, path::PathBuf, thread, time::Duration};
+use std::cell::Cell;
+
+use nakamoto::{
+    client::{
+        chan::Receiver,
+        Client,
+        Config,
+        Event,
+        Handle as ClientHandle,
+        handle::Handle,
+    },
+    net::poll::Waker,
+};
+use nakamoto::chain::Transaction;
+use nakamoto::p2p::fsm::fees::FeeEstimate;
+
+type Reactor = nakamoto::net::poll::Reactor<std::net::TcpStream>;
+
+pub struct CbfBlockchain {
+    receiver: Receiver<Event>,
+    client_handle: ClientHandle<Waker>,
+    timeout: Duration,
+    fee_data: Cell<HashMap<u32, FeeEstimate>>,
+    broadcasted_txs: Cell<Vec<Transaction>>,
+    last_sync_height: Cell<u32>,
+}
+
+pub enum CbfSyncError {
+    FilterHeaderRetrievalError,
+    BlockFilterDownloadError,
+    TransactionRetrievalError,
+}
+
+impl From<nakamoto::client::Error> for CbfSyncError {
+    fn from(_: nakamoto::client::Error) -> Self {
+        CbfSyncError::FilterHeaderRetrievalError
```

Comment:
Pass on the actual nakamoto client error value. Or else this will shoot this generic message for every error originating from nakamoto. It will be a nightmare to debug later.